File: datasets/lua-keep-incoder_6b_temp_0.2/HumanEval_52_below_threshold.lua
-- Return True if all numbers in the list l are below threshold t.
-- >>> below_threshold([1, 2, 4, 10], 100)
-- True
-- >>> below_threshold([1, 20, 4, 10], 5)
-- False
local function below_threshold(l, t)
  local n = 1
  for _, v in ipairs(l) do
    if v < t then
      return false
    end
    n = n * 10 + v
  end
  return n == t
end

-- Returns true if all numbers in the list l are above threshold t.
-- >>> above_threshold([1, 2, 4, 10], 100)
-- True
-- >>> above_threshold([1, 20, 4, 10], 5)
-- False

lu = require('luaunit')

function test_humaneval()
local candidate = below_threshold
    lu.assertEquals(candidate({1, 2, 4, 10}, 100), true)
    lu.assertEquals(candidate({1, 20, 4, 10}, 5), false)
    lu.assertEquals(candidate({1, 20, 4, 10}, 21), true)
    lu.assertEquals(candidate({1, 20, 4, 10}, 22), true)
    lu.assertEquals(candidate({1, 8, 4, 10}, 11), true)
    lu.assertEquals(candidate({1, 8, 4, 10}, 10), false)
end

os.exit(lu.LuaUnit.run())

File: datasets/R-keep-code_davinci_001_temp_0.2/HumanEval_52_below_threshold.R
# Return True if all numbers in the list l are below threshold t.
-- >>> below_threshold([1, 2, 4, 10], 100)
-- True
-- >>> below_threshold([1, 20, 4, 10], 5)
-- False
below_threshold <- function(l, t)
{
    for (i in l)
    {
        if (i > t)
        {
            return(FALSE)
        }
    }
    return(TRUE)

test_humaneval <- function() {
candidate <- below_threshold
    if(!identical(candidate(c(1, 2, 4, 10), 100), true)){quit('no', 1)}
    if(!identical(candidate(c(1, 20, 4, 10), 5), false)){quit('no', 1)}
    if(!identical(candidate(c(1, 20, 4, 10), 21), true)){quit('no', 1)}
    if(!identical(candidate(c(1, 20, 4, 10), 22), true)){quit('no', 1)}
    if(!identical(candidate(c(1, 8, 4, 10), 11), true)){quit('no', 1)}
    if(!identical(candidate(c(1, 8, 4, 10), 10), false)){quit('no', 1)}
}